### DISCUSSION TITLE
Convert string to rune slice

### DISCUSSION THREAD

**Discussion by pgundlach** [dop251/goja#616] [Q&A] [answered]
With the following code I get an error:

    TypeError: could not convert function call parameter 0: could not convert Hello to []int32 at 
      reflect.methodValueCall (native)
    exit status 255

My goal is to use a method I cannot change. How would I work around this issue? I guess (as always) I am missing something obvious.

```go
package main

import (
	"fmt"
	"os"

	"github.com/dop251/goja"
)

type document struct {
}

func (d *document) Foo(r []rune) {
	fmt.Println(r)
}

func newDocument() *document {
	d := &document{}
	return d
}

func main() {
	runtime := goja.New()

	gjNewDoc := func(call goja.FunctionCall) goja.Value {
		v := runtime.ToValue(newDocument())
		return v
	}

	runtime.Set("newdoc", gjNewDoc)

	_, err := runtime.RunString(`
var d = newdoc();
d.Foo("Hello")
`)
	if err != nil {
		fmt.Println(err)
		os.Exit(-1)
	}

}
```

**Comment by dop251** [comment, marked as answer]
There is no automatic conversion from `string` to `[]rune`, mainly because unfortunately `rune` is a type alias for `int32` and therefore they can't be distinguished at runtime.  And if your function expects`[]int32` it does not necessarily mean it expects `[]rune`.

In your specific case you could expose a method to convert a string to []rune, i.e.

```go
runtime.Set("toRunes", func toRunes(s string) []rune {
    return []rune(s)
})
// ...

_, err := runtime.RunString(`
var d = newdoc();
d.Foo(toRunes("Hello"))
`)
```

**Reply by pgundlach** [reply]
I see, thank you very much!
